File: examen final/src C/2/statistiques.c
#include "statistiques.h"

void generer_statistiques(Statistiques* statistiques, GestionnaireFilmsJeux* gestionnaire) {
    statistiques->nombre_films = 0;
    statistiques->nombre_jeux = 0;
    statistiques->total_acteurs = 0;
    statistiques->total_realisateurs = 0;

    for (int i = 0; i < gestionnaire->nombre_films_jeux; i++) {
        FilmJeu film_jeu = gestionnaire->films_jeux[i];

        if (strcmp(film_jeu.genre, "Film") == 0) {
            statistiques->nombre_films++;
        } else if (strcmp(film_jeu.genre, "Jeu") == 0) {
            statistiques->nombre_jeux++;
        }

        statistiques->total_acteurs += film_jeu.nombre_acteurs;
        statistiques->total_realisateurs += film_jeu.nombre_realisateurs;
    }

    printf("Statistiques :\n");
    printf("Nombre de films : %d\n", statistiques->nombre_films);
    printf("Nombre de jeux : %d\n", statistiques->nombre_jeux);
    printf("Total d'acteurs : %d\n", statistiques->total_acteurs);
    printf("Total de réalisateurs : %d\n", statistiques->total_realisateurs);
}

File: examen final/src PY/2/statistiques.py
class Statistiques:
    def generer_statistiques(self, gestionnaire):
        print("Statistiques :")
        print("Nombre total de films et jeux :", len(gestionnaire.films_jeux))

        if len(gestionnaire.films_jeux) > 0:
            moyenne_notes = sum([film_jeu.note_utilisateur for film_jeu in gestionnaire.films_jeux]) / len(gestionnaire.films_jeux)
            print("Moyenne des notes des utilisateurs :", moyenne_notes)
        else:
            print("Aucun film ou jeu disponible pour calculer la moyenne des notes.")
